### ISSUE TITLE
[jest/valid-describe] "First argument must be name" incorrect error

### ISSUE THREAD

**Issue by TheHolyWaffle** [jest-community/eslint-plugin-jest#203]
```
import { MY_REDUX_ACTION } from "./actions";

describe(MY_REDUX_ACTION, () => {
	//...
});
```

We're getting the error "First argument must be name" [jest/valid-describe]. This appears to be because the first argument is a variable instead of a string. 

However, our use case is testing redux store actions. Those action types are exported as strings. We use those strings to name our jest tests. 

This seems like a reasonable situation that is incorrectly marked as wrong.

**Comment by macklinu** [comment]
That use case makes sense - a PR to fix would be very welcome! 🙏

**Comment by dashmug** [comment]
Workaround...

```js
describe(`${MY_REDUX_ACTION}`, () => {
	//...
});
```

**Comment by sergioregueira** [comment]
The same happens with classes and methods:
```js
describe(Foo.name, () => {});
describe(Foo.build.name, () => {});
```
As @dashmug said, you can use templates until this issue is fixed:
```js
describe(`{Foo.name}`, () => {});
describe(`{Foo.build.name}`, () => {});
```

**Comment by chrisdothtml** [comment]
PR here: #253

**Comment by kasvtv** [comment]
This also occurs when one wants to linewrap using string concatenation like so:

```js
describe('Lorem ipsum'
    + ' dolor sit amet', () => {

});
```

**Comment by thisissami** [comment]
Hi - just chiming in here, and copying what I wrote in PR #253. tl;dr - this feels like a code smell, and this rule should probably be removed completely. Quoting myself:

Seeing as how this hasn't been merged in yet, I'm just gonna chime in with a different opinion: the `linter` should not enforce a particular form of strings, period. The first argument for a `describe` should be a string, and _any_ valid way of generating a string there should be valid. If there is no way to do this, then this should not be the linter's responsibility, and this rule should be completely removed. 

If people want typechecking, they can use `typescript`, which `jest` fully supports; it will complain if you don't pass in a string. 

If the community insists on this being a rule for the linter, then the message that it expresses should make it really clear that this might not actually be an issue. E.g. instead of 

> First argument must be name

it could be

> First argument must be name. Also if you are using a string variable or function that returns a string, then please ignore this message.

^if something like that is needed, then... it's a code smell and probably shouldn't be something the linter handles. 

My $.02.

**Comment by G-Rath** [comment]
I'm going to close this as you can use `ignoreTypeOfDescribeName` to have `valid-title` ignore this pattern.

**Comment by chrisbobbe** [comment]
Thanks for the tip! For posterity, I'll note that, as of version 23, it looks like `valid-describe` doesn't validate the title at all anymore; the way to do that (with whatever options you like) is in `valid-title`.

https://github.com/jest-community/eslint-plugin-jest/releases/tag/v23.0.0

**Comment by chrisbobbe** [comment]
> I'm going to close this as you can use `ignoreTypeOfDescribeName` to have `valid-title` ignore this pattern.

Hmm, though it looks like this only applies to `describe` blocks, not `test` or `it` blocks. Looks like getting it to work in those other cases would be difficult: https://github.com/jest-community/eslint-plugin-jest/issues/470#issuecomment-551989381.
